tests/apps/job-publisher/app.go

```diff
@@ -58,8 +58,10 @@ func main() {
 		err := publishMessagesToPubsub()
 		if err != nil {
 			log.Printf("Unable to publish, retrying.")
-			time.Sleep(2 * time.Second)
+			time.Sleep(10 * time.Second)
 		} else {
+			// Wait for a minute before shutting down to give time for any validation by E2E test code.
+			time.Sleep(1 * time.Minute)
```

Comment:
We talked about this on the call this morning, but commenting for the sake of posterity. Did we add this specifically so that our ValidateSidecar() function works correctly? Or will a user running jobs that take <1m experience some kind of failure state? If it is the latter, please make sure we have an issue so we don't lose track of it.

Reply:
Validation of sidecar is also made possible with this change. But, the reason is (like you said) to overcome an issue where K8s would not report the Job as completed on Windows.

Reply:
Issue: https://github.com/dapr/dapr/issues/2995

Reply:
I am actually thinking if this is a side effect of the Dapr sidecar being injected and then shutdown by the app and how K8s might interpret that as a failure. The shutdown code for sidecar does an exit(0), so it would be something related to timing maybe: https://github.com/dapr/dapr/blob/8fc2332dea0e5ec07333e24a942d7d54848c2c65/pkg/runtime/runtime.go#L1498